indexing-service/src/main/java/org/apache/druid/indexing/common/TaskLock.java

```diff
@@ -62,4 +63,17 @@ public interface TaskLock
   boolean isRevoked();
 
   boolean conflict(LockRequest request);
+
+  /**
+   * Checks if the lock is revoked and throws a {@link LockAcquisitionFailedException} if so.
+   *
+   * @param interval The time interval for which the lock is acquired.
+   * @throws LockAcquisitionFailedException if the lock is revoked.
+   */
+  default void assertNotRevoked(Interval interval)
```

Comment:
It feels a little weird to pass the `interval` as an argument just to include it in the exception message.
Should we use `TaskLock.getInterval()` itself instead?

Reply:
That probably makes a lot more sense.